### PR TITLE
Linting and testing TS integration

### PR DESCRIPTION
**Pull request by willvedd**
## ✏️ Changes

The initial Typescript tooling PR (#426) only covered the build step. However, it was soon discovered that our linting and unit testing processes are incompatible with Typescript. For that reason, had to make a number of small changes (highlighted below) to ensure that all our scripts function as expected for both `.js` and `.ts` file extensions.

There are several test files that needed to be changed to comply with existing conventions and be targetable by path identifiers. They were changed from `foo.tests.js` to `foo.test.js`.

One benefit here is that we were able to completely remove Babel as a dependency and instead rely almost completely on Typescript and thing TS wrappers.

## 🎯 Testing

No unit tests necessary as this just _runs_ the unit tests.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,96 +1,17 @@
 {
   "compilerOptions": {
-    /* Visit https://aka.ms/tsconfig.json to read more about this file */
```
Comment (willvedd) [inline tsconfig.json:3]: Comments are not valid JSON and as such caused the `ts-mocha` dependency to choke.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "lint:fix": "eslint --fix --ignore-path .eslintignore --ignore-pattern webpack . && kacl lint",
     "lint": "eslint --ignore-path .eslintignore --ignore-pattern webpack . && kacl lint",
     "pretest": "rimraf ./.nyc_output",
-    "test": "cross-env NODE_ENV=test nyc mocha --recursive --require @babel/register test",
+    "test": "ts-mocha -p tsconfig.json --recursive 'test/**/*.test.*'",
```
Comment (willvedd) [inline package.json:13]: From what I could tell, the `cross-env` and `nyc` dependencies weren't providing any meaningful functionality. My observations seemed to indicate that these two were functionally identical.

File: .eslintrc [hunk 3]
```diff
@@ -15,7 +15,12 @@
   },
   "settings": {
     "import/resolver": {
-      "babel-module": {}
+      "node": {
+        "extensions": [
+          ".js",
+          ".ts"
+        ]
+      }
```
Comment (willvedd) [inline .eslintrc:23]: This is to quell eslint warnings about being unable to resolve modules.

File: .eslintrc [hunk 4]
```diff
@@ -38,17 +58,46 @@
...
+      "error",
+      "ignorePackages",
+      {
+        "js": "never",
+        "ts": "never"
+      }
+    ]
   }
-}
+}
```
Comment (sergiughf) [inline .eslintrc:103]: Missing new line. 🙃

File: tsconfig.json [hunk 5]
```diff
@@ -1,96 +1,17 @@
...
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": false,
+    "skipLibCheck": true
   },
   "include": [
     "src/**/*"
   ]
-}
+}
```
Comment (sergiughf) [inline tsconfig.json:17]: Missing new line. 🙃

**Review by sergiughf — CHANGES_REQUESTED** [review]

**Pull request merged**